lib/saluki-config/src/refresher.rs

```diff
@@ -0,0 +1,107 @@
+use std::sync::Arc;
+
+use crate::{ConfigurationError, GenericConfiguration};
+use arc_swap::ArcSwap;
+use reqwest;
+use saluki_error::GenericError;
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_json::Value;
+use tokio::time::{sleep, Duration};
+
+const DATADOG_AGENT_CONFIG_ENDPOINT: &str = "https://localhost:5004/config/v1/";
+
+/// Configuration for setting up `RefreshableConfiguration`.
+#[derive(Default, Deserialize)]
+pub struct RefresherConfiguration {
+    auth_token_file_path: String,
+}
+
+/// The most recent configuration retrieved from the datadog-agent.
```

Comment:
I realize this is indeed very specific to how we're going to depend on the Datadog Agent, but let's go with a more generic description, like this:

```suggestion
/// A configuration whose values are refreshed from a remote source at runtime.
```